### ISSUE TITLE
No password prompt on reboot after locking SSD(Crucial MX500) with sedutil-cli, Error: is OPAL Failed.

### ISSUE THREAD

**Issue by ungular** [ChubbyAnt/sedutil#59]
- On reboot, after activating the SSD(Crucial MX500) locking with the "sedutil-cli" tool, the unlocking password prompt does not appear.
- the locking process goes as expected, following the steps in the instructions on the page https://sedutil.com
- checking PBA again with "linuxpba" from "sedutil-cli", the SSD appears as "is OPAL Failed".
- after removing OPAL with "reverttper" command and repeat the locking procedure,  after running the "sedutil-cli --setlockingrange 0 lk debug /dev/drive" command, linuxpba test returns "is OPAL Failed" for our SSD.

**Comment by Blacklands** [comment]
The `MX500` definitely works with `sedutil`, I've been using multiple of them in multiple systems for years. I assume you're trying to boot from it using the Shadow MBR and the PBA image? That _should_ work fine, although it seems that some hardware configurations (motherboard and its BIOS mostly?) might have issues with that?

Also, if you want to completely reset your drive and start over, try PSID reverting it (this should erase all your data, be aware of that!).

**Comment by ungular** [comment]
> The `MX500` definitely works with `sedutil`, I've been using multiple of them in multiple systems for years. I assume you're trying to boot from it using the Shadow MBR and the PBA image? That _should_ work fine, although it seems that some hardware configurations (motherboard and its BIOS mostly?) might have issues with that?
> 
> Also, if you want to completely reset your drive and start over, try PSID reverting it (this should erase all your data, be aware of that!).

loading pba as follow:
```
gunzip /usr/sedutil/UEFI64-1.15.img.gz 
#sedutil-cli --loadpbaimage debug /usr/sedutil/UEFI64-1.15.img /dev/drive 
```

how BIOS should be configured? now it's UEFI , secure boot: Off

what if i skip this command `setlockingrange 0 lk...`?

**Comment by ungular** [comment]
1. locked with sedutil-cli
2. reboot -> no password prompt
3. open again sedutil -> run: `sedutil-cli --query` -> result:
`Locking function (0x0002)
Locked=Y, LockingEnabled=Y, LockingSupported=Y, MBRDone=N, MBREnabled=Y, MBRAbsent=N, MediaEncrypt=Y`

**Comment by Blacklands** [comment]
I don't think you need a TPM for this. Where did you find that written? First time I'm hearing it I think.

So can you unlock the drive via `sedutil` in the terminal? And just booting doesn't work?

Some systems apparently have issues with the boot process, for example some end up power-cycling the drive on a reboot which just locks it again (that doesn't seem to be the case here though?).
You can get `is OPAL Failed` for multiple reasons, including just typing a wrong password.

Also, Secure Boot sadly isn't supported so yeah that needs to be turned off.

**Comment by ungular** [comment]
> I don't think you need a TPM (we're talking about a Trusted Platform Module, right?) for this. Where did you find that written? First time I'm hearing it I think.

yes, TPM 2.0. found a suggestion online.

> So can you unlock the drive via `sedutil` in the terminal? Have you tried that? And just during the boot process with the PBA it doesn't work?
>

i'm able to unlock the ssd successfully via `sedutil`. 
also for initial setup all the commands runs successfully. poweroff. 
at boot, password prompt does not appear. 
when test again with linuxpba shows "is OPAL Failed" for ssd via sedutil.

> Some systems apparently have issues with the boot process, for example some end up power-cycling the drive on a reboot which just locks it again (that doesn't seem to be the case here though?). 

so i'm going to research this now.
